server/src/test/java/org/opensearch/action/support/replication/ReplicationModeAwareProxyTests.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.support.replication;
+
+import org.opensearch.cluster.node.DiscoveryNodes;
+import org.opensearch.cluster.routing.AllocationId;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.ShardRoutingState;
+import org.opensearch.cluster.routing.TestShardRouting;
+import org.opensearch.core.index.Index;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.index.shard.IndexShardTestUtils;
+import org.opensearch.test.OpenSearchTestCase;
+
+import static org.mockito.Mockito.mock;
+
+public class ReplicationModeAwareProxyTests extends OpenSearchTestCase {
+    public void testDetermineReplicationModeTargetRoutingCurrentPrimary() {
+        ShardRouting targetRouting = TestShardRouting.newShardRouting(
+            new ShardId(new Index("test_index", "_na_"), 0),
+            "dummy-node",
+            null,
+            true,
+            ShardRoutingState.STARTED,
+            AllocationId.newInitializing("abc")
+        );
+        ShardRouting primaryRouting = TestShardRouting.newShardRouting(
+            new ShardId(new Index("test_index", "_na_"), 0),
+            "dummy-node",
+            null,
+            true,
+            ShardRoutingState.STARTED,
+            AllocationId.newInitializing("abc")
+        );
+        final ReplicationModeAwareProxy replicationModeAwareProxy = new ReplicationModeAwareProxy(
+            ReplicationMode.NO_REPLICATION,
+            DiscoveryNodes.builder().add(IndexShardTestUtils.getFakeRemoteEnabledNode("dummy-node")).build(),
+            mock(TransportReplicationAction.ReplicasProxy.class),
+            mock(TransportReplicationAction.ReplicasProxy.class)
+        );
+        assertEquals(ReplicationMode.NO_REPLICATION, replicationModeAwareProxy.determineReplicationMode(targetRouting, primaryRouting));
```

Comment:
The test appears to be relocation of primary shard. Shouldn't this have full replication mode here?